### ISSUE TITLE
precice 2.1.1 error with gcc 7.4.0 on POWER9 Summit ORNL due to PETSc broken install

### ISSUE THREAD

**Issue by luszczek** [xsdk-project/xsdk-issues#129]
I get an error related to PETSc build saying `PETSc could not be used, maybe the install is broken.`
[precise2.1.1.tar.gz](https://github.com/xsdk-project/xsdk-issues/files/5484392/precise2.1.1.tar.gz)

**Comment by luszczek** [comment]
According to `CMakeError.log` this is due to CMake using C++ compiler not MPI C++ compiler and the linker fails:`/usr/bin/ld: CMakeFiles/cmTC_29869.dir/src.cxx.o: undefined reference to symbol 'ompi_mpi_comm_self'`

**Comment by balay** [comment]
For some reason we have the following in `xsdk/package.py`
```
     depends_on('precice ~petsc', when='platform=cray +precice')
```

Perhaps we need something equivalent for summit [assuming using `'precice ~petsc'` is the fix for this issue]

**Comment by fsimonis** [comment]
> According to `CMakeError.log` this is due to CMake using C++ compiler not MPI C++ compiler and the linker fails:`/usr/bin/ld: CMakeFiles/cmTC_29869.dir/src.cxx.o: undefined reference to symbol 'ompi_mpi_comm_self'`

This is a common issue with the FindPETSc module used by this version of preCICE.

> For some reason we have the following in xsdk/package.py
>      depends_on('precice ~petsc', when='platform=cray +precice')

We had similar issues on cray back when we tested xSDK `0.5.0`, which is why we disabled it.

The most pragmatic solution for xSDK version `0.6.0` is to use `precice ~petsc` for summit as well.

We are currently committed to developing a custom FindPETSc module which uses pkg-config to detect the PETSc installation. This will be part of the next release, which is scheduled for the end of January. 
So, it won't make it into `0.6.0`, but future versions should not have this issue.
